### PR TITLE
Simplify result format by not using 'error' property

### PR DESCRIPTION
**Pull request by platfowner**
Change summary:
- Simplify result format by not using 'error' property e.g.

Before:
```
      {
        result: null,
        error: {
          code: 30001,
          message: 'Invalid get operation'
        }
      }
```
After:
```
      {
        result: null,
        code: 30001,
        message: 'Invalid get operation'
      }
```
- Rename and move SET_EXCEEDS_OP_LIST_SIZE_LIMIT from JsonRpcApiResultCode to TxResultCode as it's a part of tx result
- Remove some duplicated result code

Related issues:
- https://github.com/ainblockchain/ain-blockchain/issues/1053

### CODE REVIEW

File: common/result-code.js [hunk 1]
```diff
@@ -155,7 +160,7 @@ const JsonRpcApiResultCode = {
   GET_EXCEEDS_MAX_BYTES: 30002,
   GET_EXCEEDS_MAX_SIBLINGS: 30003,
   GET_EXCEEDS_OP_LIST_SIZE_LIMIT: 30004,
-  SET_EXCEEDS_OP_LIST_SIZE_LIMIT: 30005,
+  SET_EXCEEDS_OP_LIST_SIZE_LIMIT: 30005,  // Moved to TxResultCode.TX_SET_EXCEEDS_OP_LIST_SIZE_LIMIT
```
Comment (minsulee2) [inline common/result-code.js:163]: Should it still be kept here?

File: db/index.js [hunk 2]
```diff
@@ -788,21 +785,17 @@ class DB {
     if (!CommonUtil.isArray(opList) || CommonUtil.isEmpty(opList)) {
       return {
         result: null,
-        error: {
-          code: JsonRpcApiResultCode.GET_INVALID_OP_LIST,
-          message: `Invalid op_list given`
-        },
+        code: JsonRpcApiResultCode.GET_INVALID_OP_LIST,
```
Comment (minsulee2) [inline db/index.js:788]: I like it !! :):):)

File: json_rpc/admin.js [hunk 3]
```diff
@@ -60,10 +56,8 @@ module.exports = function getAdminApis(node) {
         const latency = Date.now() - beginTime;
         trafficStatsManager.addEvent(TrafficEventTypes.ACCESS_CONTROL_SET, latency);
         done(null, JsonRpcUtil.addProtocolVersion({
-          result: {
-            code: JsonRpcApiResultCode.ADMIN_FORBIDDEN_REQUEST,
-            message: `Forbidden request.`
-          }
+          code: JsonRpcApiResultCode.ADMIN_FORBIDDEN_REQUEST,
+          message: `Forbidden request.`
```
Comment (minsulee2) [inline json_rpc/admin.js:60]: What about using ' rather than `

File: json_rpc/admin.js [hunk 4]
```diff
@@ -85,10 +77,8 @@ module.exports = function getAdminApis(node) {
         const latency = Date.now() - beginTime;
         trafficStatsManager.addEvent(TrafficEventTypes.ACCESS_CONTROL_SET, latency);
         done(null, JsonRpcUtil.addProtocolVersion({
-          result: {
-            code: JsonRpcApiResultCode.ADMIN_VALUE_NOT_A_STRING_TYPE,
-            message: `(${args.message.value}) is not a string type.)}`
-          }
+          code: JsonRpcApiResultCode.ADMIN_VALUE_NOT_A_STRING_TYPE,
+          message: `(${args.message.value}) is not a string type.)}`
```
Comment (minsulee2) [inline json_rpc/admin.js:81]: } at the end ?!

File: json_rpc/admin.js [hunk 5]
```diff
@@ -112,10 +100,8 @@ module.exports = function getAdminApis(node) {
         const latency = Date.now() - beginTime;
         trafficStatsManager.addEvent(TrafficEventTypes.ACCESS_CONTROL_SET, latency);
         done(null, JsonRpcUtil.addProtocolVersion({
-          result: {
-            code: JsonRpcApiResultCode.ADMIN_FORBIDDEN_REQUEST,
-            message: `Forbidden request.`
-          }
+          code: JsonRpcApiResultCode.ADMIN_FORBIDDEN_REQUEST,
+          message: `Forbidden request.`
```
Comment (minsulee2) [inline json_rpc/admin.js:104]: ditto

File: json_rpc/admin.js [hunk 6]
```diff
@@ -149,10 +131,8 @@ module.exports = function getAdminApis(node) {
         const latency = Date.now() - beginTime;
         trafficStatsManager.addEvent(TrafficEventTypes.ACCESS_CONTROL_SET, latency);
         done(null, JsonRpcUtil.addProtocolVersion({
-          result: {
-            code: JsonRpcApiResultCode.ADMIN_VALUE_NOT_A_STRING_TYPE,
-            message: `(${args.message.value}) is not a string type.)}`
-          }
+          code: JsonRpcApiResultCode.ADMIN_VALUE_NOT_A_STRING_TYPE,
+          message: `(${args.message.value}) is not a string type.)}`
```
Comment (minsulee2) [inline json_rpc/admin.js:135]: ditto

File: json_rpc/admin.js [hunk 7]
```diff
@@ -195,10 +171,8 @@ module.exports = function getAdminApis(node) {
         const latency = Date.now() - beginTime;
         trafficStatsManager.addEvent(TrafficEventTypes.ACCESS_CONTROL_SET, latency);
         done(null, JsonRpcUtil.addProtocolVersion({
-          result: {
-            code: JsonRpcApiResultCode.ADMIN_FORBIDDEN_REQUEST,
-            message: `Forbidden request.`
-          }
+          code: JsonRpcApiResultCode.ADMIN_FORBIDDEN_REQUEST,
+          message: `Forbidden request.`
```
Comment (minsulee2) [inline json_rpc/admin.js:175]: ditto

File: json_rpc/admin.js [hunk 8]
```diff
@@ -232,10 +202,8 @@ module.exports = function getAdminApis(node) {
         const latency = Date.now() - beginTime;
         trafficStatsManager.addEvent(TrafficEventTypes.ACCESS_CONTROL_SET, latency);
         done(null, JsonRpcUtil.addProtocolVersion({
-          result: {
-            code: JsonRpcApiResultCode.ADMIN_VALUE_NOT_A_STRING_TYPE,
-            message: `(${args.message.value}) is not a string type.)}`
-          }
+          code: JsonRpcApiResultCode.ADMIN_VALUE_NOT_A_STRING_TYPE,
+          message: `(${args.message.value}) is not a string type.)}`
```
Comment (minsulee2) [inline json_rpc/admin.js:206]: ditto

File: json_rpc/transaction.js [hunk 9]
```diff
@@ -82,19 +82,17 @@ module.exports = function getTransactionApis(node, p2pServer) {
...
         const latency = Date.now() - beginTime;
         trafficStatsManager.addEvent(TrafficEventTypes.JSON_RPC_SET, latency);
         done(null, JsonRpcUtil.addProtocolVersion({
-          result: {
-            code: JsonRpcApiResultCode.TX_MISSING_PROPERTIES,
-            message: `Missing properties.`
-          }
+          result: null,
+          code: JsonRpcApiResultCode.TX_MISSING_PROPERTIES,
+          message: `Missing properties.`
```
Comment (minsulee2) [inline json_rpc/transaction.js:95]: ditto

File: json_rpc/transaction.js [hunk 10]
```diff
@@ -103,20 +101,18 @@ module.exports = function getTransactionApis(node, p2pServer) {
           const latency = Date.now() - beginTime;
           trafficStatsManager.addEvent(TrafficEventTypes.JSON_RPC_SET, latency);
           done(null, JsonRpcUtil.addProtocolVersion({
-            result: {
-              code: JsonRpcApiResultCode.TX_INVALID_FORMAT,
-              message: `Invalid transaction format.`
-            }
+            result: null,
+            code: JsonRpcApiResultCode.TX_INVALID_FORMAT,
+            message: `Invalid transaction format.`
```
Comment (minsulee2) [inline json_rpc/transaction.js:106]: ditto

File: json_rpc/transaction.js [hunk 11]
```diff
@@ -103,20 +101,18 @@ module.exports = function getTransactionApis(node, p2pServer) {
...
               NodeConfigs.ENABLE_EARLY_TX_SIG_VERIF &&
               !Transaction.verifyTransaction(createdTx, chainId)) {
             done(null, JsonRpcUtil.addProtocolVersion({
-              result: {
-                code: JsonRpcApiResultCode.TX_INVALID_SIGNATURE,
-                message: `Invalid transaction signature.`
-              }
+              result: null,
+              code: JsonRpcApiResultCode.TX_INVALID_SIGNATURE,
+              message: `Invalid transaction signature.`
```
Comment (minsulee2) [inline json_rpc/transaction.js:115]: ditto

File: json_rpc/transaction.js [hunk 12]
```diff
@@ -135,19 +131,17 @@ module.exports = function getTransactionApis(node, p2pServer) {
         const latency = Date.now() - beginTime;
         trafficStatsManager.addEvent(TrafficEventTypes.JSON_RPC_SET, latency);
         done(null, JsonRpcUtil.addProtocolVersion({
-          result: {
-            code: JsonRpcApiResultCode.BATCH_INVALID_FORMAT,
-            message: `Invalid batch transaction format.`
-          }
+          result: null,
+          code: JsonRpcApiResultCode.BATCH_INVALID_FORMAT,
+          message: `Invalid batch transaction format.`
```
Comment (minsulee2) [inline json_rpc/transaction.js:136]: ditto

File: test/integration/node.test.js [hunk 13]
```diff
@@ -4028,10 +4072,9 @@ describe('Blockchain Node', () => {
           protoVer: BlockchainConsts.CURRENT_PROTOCOL_VERSION
         }).then((res) => {
           assert.deepEqual(res.result, {
-            result: {
-              code: 30302,
-              message: `Missing properties.`,
-            },
+            result: null,
+            code: 30302,
+            message: `Missing properties.`,
```
Comment (minsulee2) [inline test/integration/node.test.js:4077]: ditto

File: test/integration/node.test.js [hunk 14]
```diff
@@ -4057,10 +4100,9 @@ describe('Blockchain Node', () => {
           protoVer: BlockchainConsts.CURRENT_PROTOCOL_VERSION
         }).then((res) => {
           assert.deepEqual(res.result, {
-            result: {
-              code: 30303,
-              message: `Invalid transaction format.`,
-            },
+            result: null,
+            code: 30303,
+            message: `Invalid transaction format.`,
```
Comment (minsulee2) [inline test/integration/node.test.js:4105]: ditto

File: test/integration/node.test.js [hunk 15]
```diff
@@ -4086,10 +4128,9 @@ describe('Blockchain Node', () => {
           protoVer: BlockchainConsts.CURRENT_PROTOCOL_VERSION
         }).then((res) => {
           assert.deepEqual(res.result, {
-            result: {
-              code: 30304,
-              message: `Invalid transaction signature.`,
-            },
+            result: null,
+            code: 30304,
+            message: `Invalid transaction signature.`,
```
Comment (minsulee2) [inline test/integration/node.test.js:4133]: ditto

File: test/integration/node.test.js [hunk 16]
```diff
@@ -4329,16 +4370,15 @@ describe('Blockchain Node', () => {
           protoVer: BlockchainConsts.CURRENT_PROTOCOL_VERSION
         }).then((res) => {
           assert.deepEqual(res.result, {
-            result: {
-              code: 30401,
-              message: `Invalid batch transaction format.`
-            },
+            result: null,
+            code: 30401,
+            message: `Invalid batch transaction format.`,
```
Comment (minsulee2) [inline test/integration/node.test.js:4375]: ditto

File: common/result-code.js (hunk 1 above)
Comment (platfowner) [reply common/result-code.js]: Let's keep it to avoid mistakenly reused cases.
Just commented it out instead. PTAL

File: json_rpc/admin.js (hunk 3 above)
Comment (platfowner) [reply json_rpc/admin.js]: just done, thanks!

File: json_rpc/admin.js (hunk 4 above)
Comment (platfowner) [reply json_rpc/admin.js]: just fixed, thanks!

File: json_rpc/admin.js (hunk 5 above)
Comment (platfowner) [reply json_rpc/admin.js]: ditto

File: json_rpc/admin.js (hunk 6 above)
Comment (platfowner) [reply json_rpc/admin.js]: ditto

File: json_rpc/admin.js (hunk 7 above)
Comment (platfowner) [reply json_rpc/admin.js]: ditto

File: json_rpc/admin.js (hunk 8 above)
Comment (platfowner) [reply json_rpc/admin.js]: ditto

File: json_rpc/transaction.js (hunk 9 above)
Comment (platfowner) [reply json_rpc/transaction.js]: ditto

File: json_rpc/transaction.js (hunk 10 above)
Comment (platfowner) [reply json_rpc/transaction.js]: ditto

File: json_rpc/transaction.js (hunk 11 above)
Comment (platfowner) [reply json_rpc/transaction.js]: ditto

File: json_rpc/transaction.js (hunk 12 above)
Comment (platfowner) [reply json_rpc/transaction.js]: ditto

File: test/integration/node.test.js (hunk 14 above)
Comment (platfowner) [reply test/integration/node.test.js]: ditto

File: test/integration/node.test.js (hunk 13 above)
Comment (platfowner) [reply test/integration/node.test.js]: ditto

File: test/integration/node.test.js (hunk 15 above)
Comment (platfowner) [reply test/integration/node.test.js]: ditto

File: test/integration/node.test.js (hunk 16 above)
Comment (platfowner) [reply test/integration/node.test.js]: ditto

**Comment by platfowner** [comment]
Comments addressed, PTAL~ @minsulee2

File: common/result-code.js (hunk 1 above)
Comment (platfowner) [reply common/result-code.js]: Just added DEPRECATED_ prefix instead of commenting out. PTAL~

**Review by minsulee2 — APPROVED** [review]
LGTM! Thanks for the fast fix. 👍

**Comment by platfowner** [comment]
Thanks for the review. Now merging.. @minsulee2

**Pull request merged**